src/test/java/bio/terra/workspace/service/datareference/DataReferenceValidationUtilsTest.java

```diff
@@ -0,0 +1,54 @@
+package bio.terra.workspace.service.datareference;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.service.datareference.exception.InvalidDataReferenceException;
+import bio.terra.workspace.service.datareference.utils.DataReferenceValidationUtils;
+import org.junit.jupiter.api.Test;
+
+public class DataReferenceValidationUtilsTest extends BaseUnitTest {
+
+  @Test
+  public void testInvalidCharInReferenceName() {
+    assertThrows(
+        InvalidDataReferenceException.class,
+        () -> DataReferenceValidationUtils.validateReferenceName("invalid@@@name"));
+  }
+
+  @Test
+  public void validCharInReferenceNameOk() {
+    DataReferenceValidationUtils.validateReferenceName("valid_name");
+  }
+
+  @Test
+  public void testInvalidCharInBucketName() {
+    assertThrows(
+        InvalidDataReferenceException.class,
+        () -> DataReferenceValidationUtils.validateBucketName("INVALIDBUCKETNAME"));
```

Comment:
It would be nice to have this validation at a higher level so controlled resources can use it too.

Reply:
This is in a validation util class called by the controller, I think it's actually reasonable for the various controlled resource controllers to use this bean as well